File: apps/api/app/schemas/ledger.py
import uuid
from datetime import date
from decimal import Decimal

from pydantic import BaseModel


class LedgerEntry(BaseModel):
    date: date
    type: str
    reference: str
    debit: Decimal
    credit: Decimal
    balance: Decimal


class PartyStatement(BaseModel):
    party_id: uuid.UUID
    party_name: str
    opening_balance: Decimal
    closing_balance: Decimal
    entries: list[LedgerEntry]

File: apps/web/src/lib/api/ledger.ts
import { apiRequest } from "@/lib/api-client";

export interface LedgerEntry {
  date: string;
  type: string;
  reference: string;
  debit: string;
  credit: string;
  balance: string;
}

export interface PartyStatement {
  party_id: string;
  party_name: string;
  opening_balance: string;
  closing_balance: string;
  entries: LedgerEntry[];
}

export const ledgerApi = {
  customerStatement: (customerId: string) => apiRequest<PartyStatement>(`/customers/${customerId}/statement`),
  supplierStatement: (supplierId: string) => apiRequest<PartyStatement>(`/suppliers/${supplierId}/statement`),
};
